src/app/services/params.service.ts

```diff
@@ -0,0 +1,98 @@
+import { inject, Injectable } from '@angular/core';
+import { DateFormat, DateRange } from '../models';
+import { QueryParams } from '../models/query-params.model';
+import { DateService } from './date';
+
+/**
+ * Gets and sets params without changing navigation history.
+ * Because this, it doesn't work with the `Router` Service.
+ */
+@Injectable({
+	providedIn: 'root',
+})
+export class ParamsService {
+	private readonly dateService = inject(DateService);
+
+	/**
+	 * Gets param value by name
+	 */
+	getParam(name: string): string | null {
+		try {
+			const params = new URLSearchParams(window.location.search);
```

Comment:
Should we be concerned with browser support for `URLSearchParams`? The params functionality isn't _requied_ for the application to run properly, so I'm fine with them erroring silently, but if there isn't an actual risk, I'd like to either remove the try/catch logic or just not add anymore related to `URLSearchParams`

Reply:
+1 for Router for Denison

Reply:
https://angular.io/api/router/NavigationBehaviorOptions#skipLocationChange

Reply:
This comment was meant to be for the section below (same Service though)